### DISCUSSION TITLE
Exemplo syntax AWS S3 sinks

### DISCUSSION THREAD

**Discussion by Douglaso** [vectordotdev/vector#14907] [General]
I saw all doc about sinks to AWS S3, but I don`t understud how do put "AWS_ACCESS_KEY_ID" and "AWS_SECRET_ACCESS_KEY" on sinks in vector.toml. You can give me a exemplo about syntax to use it.
Thank you.

**Comment by jszwedko** [comment]
Hi @Douglaso !

Are you trying to configure static credentials for the sink? If so, you can set:

```
auth.access_key_id = "..."
auth.secret_access_key = "..."
```

You can also provide those values via the standard AWS environment variables like `$AWS_ACCESS_KEY_ID` and `$AWS_SECRET_ACCESS_KEY` which the AWS SDK Vector uses will read automatically.

**Reply by Douglaso** [reply]
Hi @jszwedko! tks for help me. but I`m putting this. And continuios receveid error. Follow print of the config and error.

![image](https://user-images.githubusercontent.com/7070020/197054779-b8dc9fc3-e6cc-4bbb-a1af-1a6b1302c052.png)

**Reply by jszwedko** [reply]
That could indicate that the credentials you provided don't have access to `ListBucket` the targeted bucket.

**Reply by Douglaso** [reply]
ok, i thought of that too! however I installed the aws cli on linux and by the command "aws s3 ls" I can list the bucket. So I believe the credentials are correct and the permission is valid. Would you have any other suggestions? please.

**Reply by jszwedko** [reply]
My next question is if they are using the same credentials. You can get some debug AWS logs by running Vector with the following environment variable:

```
export VECTOR_LOG=aws_config=debug
```

That might give you some more clues.

**Reply by Douglaso** [reply]
Tks so much! with the debug I found error and apply the fix. What can be this  message in log?

{"appname":"devankoshal","facility":"kern","hostname":"names.net","message":"There's a breach in the warp core, captain","msgid":"ID218","procid":3799,"severity":"warning","timestamp":"2022-10-21T17:38:31.964Z","version":2}

**Reply by jszwedko** [reply]
Hi @Douglaso ! Glad the fix worked. I'm not sure I understand the second question. Could you rephrase it?

**Reply by Douglaso** [reply]
Hi, when I started vector service in log archive has this error. Do you know what can be?

ERROR:
Oct 25 14:11:51 dcbvm073pr101 vector[846729]: 2022-10-25T17:11:51.546386Z ERROR source{component_kind="source" component_id=networks_logs component_type=file component_name=networks_logs}:file_server: vector::internal_events::file::source: Failed reading file for fingerprinting. file=/var/log/unicred/network/switchs-usuarios.log-20221025.gz error=Permission denied (os error 13) error_code="reading_fingerprint" error_type="reader_failed" stage="receiving"


{"appname":"shaneIxD","facility":"alert","hostname":"make.org","message":"We're gonna need a bigger boat","msgid":"ID307","procid":2601,"severity":"emerg","timestamp":"2022-10-25T17:13:21.284Z","version":1}

**Reply by jszwedko** [reply]
Hi @Douglaso ! That error indicates that Vector was unable to read that file due to a lack of permissions.

**Reply by Douglaso** [reply]
It is a .gz file, is it not possible to upload a .gz file?

**Reply by jszwedko** [reply]
It is possible. The error indicates that the OS user that is running the Vector process could not read the referenced file because it lacked filesystem permissions.

**Reply by Douglaso** [reply]
I putted 777 of permission, but the error is continuous. Or are you talk about permission in another directory?

![image](https://user-images.githubusercontent.com/7070020/198033114-729b00c4-fbaa-4322-81a2-dc8e29e1857d.png)
